eva-server/src/main/java/uk/ac/ebi/eva/server/ws/ga4gh/beaconv2/BeaconServiceV2.java

```diff
@@ -0,0 +1,328 @@
+
+@Service
+public class BeaconServiceV2 {
+
+    @Autowired
+    private VariantWithSamplesAndAnnotationsService service;
+
+    @Autowired
+    private VariantSourceService variantSourceService;
+
+    public BeaconServiceV2() {
+    }
+
+    public BeaconImpl getBeacon() {
+        BeaconImpl beacon = new BeaconImpl();
+        beacon.setDatasets(getAllBeaconDatasets());
+        return beacon;
+    }
+
+    private List<BeaconDataset> getAllBeaconDatasets() {
+        List<BeaconDataset> beaconDatasets = new ArrayList<>();
+        beaconDatasets.addAll(getBeaconDatasetsPerDatabase("hsapiens", "GRCh37"));
+        beaconDatasets.addAll(getBeaconDatasetsPerDatabase("hsapiens", "GRCh38"));
+        return beaconDatasets;
+    }
+
+    private List<BeaconDataset> getBeaconDatasetsPerDatabase(String species, String assemblyId) {
+        List<BeaconDataset> beaconDatasets = new ArrayList<>();
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species + "_" +
+                assemblyId.toLowerCase()));
+        List<VariantSource> variantSources = variantSourceService.findAllVariantSourcesForBeacon();
+        variantSources.forEach(
+                variantSource -> beaconDatasets.add(
+                        new BeaconDataset().id(variantSource.getStudyId())
+                                .name(variantSource.getStudyName())
+                                .assemblyId(assemblyId)
+                                .updateDateTime(variantSource.getDate() == null ? null :
+                                        variantSource.getDate().toString())
+                                .sampleCount(variantSource.getStats() == null ? null :
+                                        (long) variantSource.getStats().getSamplesCount())
+                                .variantCount(variantSource.getStats() == null ? null :
+                                        (long) variantSource.getStats().getVariantsCount())));
+        return beaconDatasets;
+    }
+
+    public BeaconAlleleResponse find(String chromosome, Long start, Long startMin, Long startMax, Long end, Long endMin,
+                                     Long endMax, String referenceBases, String alternateBases, String variantType,
+                                     String assemblyId, List<String> studies, String includeDatasetResponses) {
+
+        try {
+            checkParameters(chromosome, referenceBases, start, end, alternateBases, variantType, assemblyId,
+                    includeDatasetResponses);
+        } catch (IllegalArgumentException e) {
+            return buildBeaconAlleleResponse(null, null, null, e.getMessage());
+        }
+
+        BeaconAlleleRequest request = buildBeaconAlleleRequest(chromosome, start, startMin, startMax, end, endMin,
+                endMax, referenceBases, alternateBases, variantType, assemblyId, studies, includeDatasetResponses);
+
+        if (assemblyId.equalsIgnoreCase("grch37")) {
```

Comment:
I think it could make sense to add this check to the `checkParameters` method.

Reply:
Fixed :)